ComputeClient/Compute.Contracts/Snapshot/Snapshot.cs

```diff
@@ -2437,18 +2451,14 @@ public bool replica
         }
     }
 
+
     /// <remarks/>
-    [System.Xml.Serialization.XmlAttributeAttribute()]
+    [Obsolete("archived is deprecated, please use archiveStatus instead.")]
     public bool archived
     {
         get
         {
-            return this.archivedField;
-        }
-        set
-        {
-            this.archivedField = value;
+            return !string.IsNullOrEmpty(archiveStatus) && (archiveStatus.ToUpper().Equals("ARCHIVED") || archiveStatus.ToUpper().Equals("DOWNLOAD_IN_PROGRESS"));
```

Comment:
Deriving the legacy "archived" value based on the following mapping.

archiveStatus (new field name)        -> archived (legacy field name being removed)
-----------------------------------------------------------------------------------
null                                                   -> false
AVAILABLE_LOCALLY                        -> false
ARCHIVED                                        -> true
DOWNLOAD_IN_PROGRESS             -> true